### PR TITLE
reqlog: cnonce might not be a valid utf-8 string

### PR DESCRIPTION
**Pull request by nirbhayc**
This patch keeps cson parser from panicking.

### CODE REVIEW

File: db/eventlog.c [hunk 1]
```diff
@@ -485,8 +485,13 @@ static void eventlog_add_int(cson_object *obj, const struct reqlogger *logger)
         p = &logger->iq->snap_info;
     else if (logger->clnt && get_cnonce(logger->clnt, &snap) == 0)
         p = &snap;
-    if (p)
-        cson_object_set(obj, "cnonce", cson_value_new_string(p->key, p->keylen));
+    if (p) {
+        char cnonce[MAX_SNAP_KEY_LEN + 1];
```
Comment (adizaimi) [inline db/eventlog.c]: Util_tohex() needs cnonce[ 2 * MAX_SNAP_KEY_LEN + 1]

Comment (nirbhayc) [reply db/eventlog.c]: Good catch! thanks

**Comment by nirbhayc** [comment]
Cherry-picked to 7.0.

**Comment by nirbhayc** [comment]
@akshatsikarwar Thanks for the review!

**Pull request merged**